File: test/initial/AdresseViewModel.cs

namespace SSI.Domain.ViewModels
{
    public class AdresseViewModel
    {
        public int Id { get; set; }
        public string Adresse1 { get; set; }
        public string Adresse2 { get; set; }
        public string Adresse3 { get; set; }
        public string Adresse4 { get; set; }
        public string CodePostal { get; set; }
        public string Ville { get; set; }
        public EntityItemViewModel Pays { get; set; }
        public string Email { get; set; }
        
        public EntityItemViewModel TitreAdresse { get; set; }
        public bool IsActif { get; set; }

        /*
        *   GET
        */

        //public List<EntityItemViewModel> AllPays { get; set; }
        //public List<EntityItemViewModel> AllTitresAdresse { get; set; }
    }
}

File: test/final/AdresseViewModel.ts
/// <reference path="knockout.d.ts"/>
class AdresseViewModel
{
    public Id: KnockoutObservable<number>;
    public CodePostal: KnockoutObservable<string>;
    public Ville: KnockoutObservable<string>;
    public Pays: KnockoutObservable<EntityItemViewModel>;
    public Email: KnockoutObservable<string>;
    public TitreAdresse: KnockoutObservable<EntityItemViewModel>;
    public IsActif: KnockoutObservable<boolean>;
    public AllPays: KnockoutObservableArray<EntityItemViewModel>;
    public AllTitresAdresse: KnockoutObservableArray<EntityItemViewModel>;

    constructor(){
        this.Id = ko.observable(0);
        this.CodePostal = ko.observable("");
        this.Ville = ko.observable("");
        this.Pays = ko.observable(new EntityItemViewModel());
        this.Email = ko.observable("");
        this.TitreAdresse = ko.observable(new EntityItemViewModel());
        this.IsActif = ko.observable(false);
        this.AllPays = ko.observableArray([]);
        this.AllTitresAdresse = ko.observableArray([]);
    }
}
